src/parser/SparqlTriple.h

```diff
@@ -53,6 +54,21 @@ class SparqlTripleSimple : public SparqlTripleBase<TripleComponent> {
   using Base::Base;
 };
 
+class SparqlTripleSimpleWithGraph : public SparqlTripleSimple {
+ public:
+  using SparqlTripleSimple::SparqlTripleSimple;
+  SparqlTripleSimpleWithGraph(TripleComponent s, TripleComponent p,
+                              TripleComponent o,
+                              std::variant<Iri, Variable, std::monostate> g,
+                              AdditionalScanColumns additionalScanColumns = {})
+      : SparqlTripleSimple(std::move(s), std::move(p), std::move(o),
+                           std::move(additionalScanColumns)),
+        g_{std::move(g)} {}
+  std::variant<Iri, Variable, std::monostate> g_;
```

Comment:
Should The default here be `monostate`? Then you probably have to either change the order, or be explicit here
(you inherit a constructor if i see that correctly.
Also you could give the `variant` a name, e.g `using Graph = variant<...>`.